File: backend/supabase_client.py
from supabase import create_client, Client
import os
from dotenv import load_dotenv

load_dotenv()

supabase_url = os.getenv("SUPABASE_URL")
supabase_key = os.getenv("SUPABASE_SERVICE_ROLE_KEY")

# 환경 변수가 없어도 에러를 발생시키지 않고 None으로 설정
# 사용 시점에 체크하도록 변경
if not supabase_url or not supabase_key:
    print("[경고] Supabase 환경 변수가 설정되지 않았습니다. Supabase 기능이 비활성화됩니다.")
    print("[경고] .env 파일에 SUPABASE_URL과 SUPABASE_SERVICE_ROLE_KEY를 추가하세요.")
    supabase: Client | None = None
else:
    supabase: Client = create_client(supabase_url, supabase_key)

File: frontend/src/supabaseClient.js
import { createClient } from '@supabase/supabase-js'

const supabaseUrl = process.env.REACT_APP_SUPABASE_URL
const supabaseAnonKey = process.env.REACT_APP_SUPABASE_ANON_KEY

console.log("Supabase 환경 변수 확인:");
console.log("REACT_APP_SUPABASE_URL:", supabaseUrl ? "설정됨" : "❌ 설정 안됨");
console.log("REACT_APP_SUPABASE_ANON_KEY:", supabaseAnonKey ? "설정됨" : "❌ 설정 안됨");

if (!supabaseUrl || !supabaseAnonKey) {
  console.error("⚠️ Supabase 환경 변수가 설정되지 않았습니다!");
  console.error("frontend/.env 파일을 생성하고 다음 내용을 추가하세요:");
  console.error("REACT_APP_SUPABASE_URL=https://your-project.supabase.co");
  console.error("REACT_APP_SUPABASE_ANON_KEY=your_anon_key");
  // 에러를 던지지 않고 경고만 표시 (개발 중 테스트 가능하도록)
  // throw new Error('Supabase 환경 변수가 설정되지 않았습니다.')
}

// 환경 변수가 없어도 클라이언트는 생성하되, 사용 시 에러가 발생하도록
export const supabase = supabaseUrl && supabaseAnonKey 
  ? createClient(supabaseUrl, supabaseAnonKey)
  : null